src/commands/ext-install.ts

```diff
@@ -44,75 +45,128 @@ interface InstallExtensionOptions {
+  const {
+    projectId,
+    extensionName,
+    source,
+    extVersion,
+    paramFilePath,
+    nonInteractive,
+    force,
+  } = options;
   const spec = source?.spec || extVersion?.spec;
   if (!spec) {
     throw new FirebaseError(
       `Could not find the extension.yaml for ${extensionName}. Please make sure this is a valid extension and try again.`
     );
   }
-  const spinner = ora.default(
-    "Installing your extension instance. This usually takes 3 to 5 minutes..."
-  );
+  const spinner = ora.default();
   try {
     await provisioningHelper.checkProductsProvisioned(projectId, spec);
 
     if (spec.billingRequired) {
       const enabled = await checkBillingEnabled(projectId);
-      if (!enabled) {
+      if (!enabled && nonInteractive) {
+        throw new FirebaseError(
+          `This extension requires the Blaze plan, but project ${projectId} is not on the Blaze plan. ` +
+            marked(
+              "Please visit https://console.cloud.google.com/billing/linkedaccount?project=${projectId} to upgrade your project."
+            )
+        );
+      } else if (!enabled) {
         await displayNode10CreateBillingNotice(spec, false);
         await enableBilling(projectId, spec.displayName || spec.name);
       } else {
-        await displayNode10CreateBillingNotice(spec, true);
+        await displayNode10CreateBillingNotice(spec, !nonInteractive);
       }
     }
     const roles = spec.roles ? spec.roles.map((role: extensionsApi.Role) => role.role) : [];
-    await askUserForConsent.prompt(spec.displayName || spec.name, projectId, roles);
-
+    if (roles.length) {
+      await askUserForConsent.displayRoles(spec.displayName || spec.name, projectId, roles);
+      const consented = await confirmInstallInstance(nonInteractive, force);
+      if (!consented) {
+        throw new FirebaseError(
+          "Without explicit consent for the roles listed, we cannot deploy this extension."
+        );
+      }
+    }
     let instanceId = spec.name;
+
+    let choice: "updateExisting" | "installNew" | "cancel";
     const anotherInstanceExists = await instanceIdExists(projectId, instanceId);
     if (anotherInstanceExists) {
-      const consent = await promptForRepeatInstance(projectId, spec.name);
-      if (!consent) {
-        // TODO(b/145233161): Add documentation link about extension instances here.
-        logger.info(
-          marked(
-            "Installation cancelled. For a list of all available Firebase Extensions commands, run `firebase ext`."
-          )
+      if (!nonInteractive) {
+        choice = await promptForRepeatInstance(projectId, spec.name);
+      } else if (nonInteractive && force) {
+        choice = "updateExisting";
+      } else {
+        throw new FirebaseError(
+          `An extension with the ID '${clc.bold(
+            extensionName
+          )}' already exists in the project '${clc.bold(projectId)}'.` +
```

Comment:
better to point out how to fix it too "Select a new extension instance ID in your configuration"